### ISSUE TITLE
eventsFunctionContext.returnValue do nothing in functions

### ISSUE THREAD

**Issue by Bouh** [4ian/GDevelop#1072]
## Describe the bug
It seems that the following code does not work.
`eventsFunctionContext.returnValue = true;`
If true and false are not returned maybe string and number are not returned too.

To bypass this I used a scene variable look this screen.
![image](https://user-images.githubusercontent.com/1670670/58674169-d96d0880-834e-11e9-8f78-d536048b235a.png)

[Here the project file](https://master.tus.io/files/7f6fc3eba7453ed180c0cebb41d31a36+ihEWywJigAL2j5nYH2Use56FOKT6G6Lr1hpNe_f_uZEum1Q82G9ufKszsT8tsqaoSdsffIdmhdFOpdN2qjy6SGBCFZRiqmyIIxtw4pgWtPCtJhwPcFMc93i30bPaoIbi)

## Other details
b70, win10

**Comment by Bouh** [comment]
Return value work well i close this issue, sorry.
Problem is in my JS code, when i use multiple controllers the function return alway false.

**Comment by 4ian** [comment]
I was about to answer :) You're using requestAnimationFrame, which can lead here to strange/dangerous results because it's already used by GDevelop to handle the main loop. Meaning that you would get something like this;

- GD events are running
- Your JS code is running, calling requestAnimationFrame
- GD events are running
- Your JS code is running, calling requestAnimationFrame
- Your function "gameLoop" is called, but try to change eventsFunctionContext while it's finished since a long time

etc...

basically, you have an issue because functions are supposed to be synchronous. They must do they work and call eventsFunctionContext.returnValue without calling any "asynchronous callback".
Avoid at all cost to call requestAnimationFrame in your JS code. This won't lead to good result because functions are already executed 60 frame per seconds - you are *already* inside a requestAnimationFrame ;)

**Comment by 4ian** [comment]
Check instead if it's possible to simply read the state of the gamepads without having this complex requestAnimationFrame initialization. It must be coming from an article/tutorial - but remember that in the case of GD, you are already running side the context of a game being updated 60 times a second.
